# Hypothesis

### I think that the majority of the students in COMP 110 had less than one month of programming experience prior to enrolling in the class.

# Procedure

First step: read the csv and convert it to a column-oriented table.

In [9]:
%reload_ext autoreload
%autoreload 2

from data_utils_2 import read_csv_rows, columnar

data_rows: list[dict[str, str]] = read_csv_rows('../../data/survey.csv')
columns: dict[str, list[str]] = columnar(data_rows)

Next step: I must select just the prior experience column to measure our responses. Displayed are the first 10 rows of data.

In [12]:
from data_utils_2 import head, select

subset: dict[str, list[str]] = select(columns, ["prior_exp"])
head(subset, 10)

{'prior_exp': ['None to less than one month!',
  'None to less than one month!',
  'None to less than one month!',
  'None to less than one month!',
  'None to less than one month!',
  'Over 2 years',
  'None to less than one month!',
  'None to less than one month!',
  'None to less than one month!',
  'None to less than one month!']}

Final step: Count up the totals each type of response (either less than two months, or greater than two months).

In [20]:
from data_utils_2 import count
counts = count(subset["prior_exp"])

print(f"Less than 1 month total is: {counts['None to less than one month!']} ")
print(f"More than 1 month total is: {counts['2-6 months'] + counts['7-12 months'] + counts['1-2 years'] + counts['Over 2 years']} ")

Less than 1 month total is: 403 
More than 1 month total is: 170 


A good way to represent this is through a percentage, so I will import a simple helper function that can calculate the percentage of students who had less than one month of programming experience using the values I collected in the last section.

In [35]:
from data_utils_2 import ratio

print(f"The percentage of students in COMP 110 who had less than one month of programming experience is: {ratio(403, 170)}%. ")

The percentage of students in COMP 110 who had less than one month of programming experience is: 70.33158813263525%. 


# Conclusion

### As shown by the data, more students had less than one month of programming experience prior to enrolling in COMP 110, at more than double the amount that had more than one month of programming experience. The percentage of students with less than one month of programming experience prior to taking COMP 110 is around 70.3%. This is likely due to the fact that COMP 110 is an entry-level class, designed for beginners.